client/asset/dcr/dcr.go

```diff
@@ -3096,27 +3013,28 @@ func (dcr *ExchangeWallet) createSig(tx *wire.MsgTx, idx int, pkScript []byte, a
 		return nil, nil, err
 	}
 
-	priv, pub, err := dcr.getKeys(addr)
+	priv, err := dcr.wallet.AddressPrivKey(dcr.ctx, addr)
 	if err != nil {
 		return nil, nil, err
 	}
+	defer priv.Zero()
 
 	sig, err = sign.RawTxInSignature(tx, idx, pkScript, txscript.SigHashAll, priv.Serialize(), sigType)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return sig, pub.SerializeCompressed(), nil
+	return sig, priv.PubKey().SerializeCompressed(), nil
 }
 
 // getKeys fetches the private/public key pair for the specified address.
 func (dcr *ExchangeWallet) getKeys(addr stdaddr.Address) (*secp256k1.PrivateKey, *secp256k1.PublicKey, error) {
-	wif, err := dcr.wallet.AddressPrivKey(dcr.ctx, addr)
+	priv, err := dcr.wallet.AddressPrivKey(dcr.ctx, addr)
 	if err != nil {
 		return nil, nil, fmt.Errorf("%w (is wallet locked?)", err)
 	}
+	defer priv.Zero()
 
-	priv := secp256k1.PrivKeyFromBytes(wif.PrivKey())
 	return priv, priv.PubKey(), nil
```

Comment:
Will the returned `priv` still be useful if it's zeroed immediately after it is returned?

Reply:
Whoops